src/PrizeVault.sol

```diff
@@ -777,22 +785,25 @@ contract PrizeVault is TwabERC20, Claimable, IERC4626, ILiquidationSource, Ownab
         return (false, 0);
     }
 
-    /// @notice Returns the total assets that are owed to share holders and any other internal balances.
-    /// @dev The yield fee balance is included since it's cheaper to keep track of those shares
-    ///      internally instead of doing an additional TWAB mint on every liquidation.
-    /// @param _totalSupply The total share supply of the vault
-    /// @return The total asset debt of the vault
-    function _totalDebt(uint256 _totalSupply) internal view returns (uint256) {
-        return _totalSupply + yieldFeeBalance;
+    /// @notice Returns the shares that can be minted without exceeding the TwabController supply limit.
+    /// @dev The TwabController limits the total supply for each vault.
+    /// @param _existingShares The current allocated prize vault shares (internal and external)
+    /// @return The remaining shares that can be minted without exceeding TWAB limits
+    function _mintLimit(uint256 _existingShares) internal pure returns (uint256) {
+        return TWAB_SUPPLY_LIMIT - _existingShares;
     }
 
-    /// @notice Returns the remaining supply that can be minted without exceeding the TwabController limits.
-    /// @dev The TwabController limits the total supply for each vault to uint96
-    /// @param _totalSupply The total share supply of the vault
-    /// @return The remaining supply that can be minted without exceeding TWAB limits
-    function _twabSupplyLimit(uint256 _totalSupply) internal pure returns (uint256) {
-        unchecked {
-            return type(uint96).max - _totalSupply;
+    /// @notice Verifies that the mint limit can support the new share balance.
+    /// @dev Reverts if the mint limit is exceeded.
+    /// @dev This MUST be called anytime there is a positive increase in the net total shares.
+    /// @param _existingShares The total existing prize vault shares (internal and external)
+    /// @param _newShares The new shares
+    function _enforceMintLimit(uint256 _existingShares, uint256 _newShares) internal pure {
```

Comment:
It's odd that the param here is "newShares", but in usage you actually pass "newYieldVaultFee", which is a combination of new + old, and the totalsupply as existing shares.